packages/frontend/src/components/common/Table/getGroupedRowModelLightdash.ts

```diff
@@ -0,0 +1,225 @@
+import { createRow } from '@tanstack/table-core/src/core/row';
+import {
+    type Row,
+    type RowData,
+    type RowModel,
+    type Table,
+} from '@tanstack/table-core/src/types';
+import { flattenBy, memo } from '@tanstack/table-core/src/utils';
+
+// Customizes row grouping based on a clone of
+// @tanstack/table-core/src/utils/getGroupedRowModel.ts
+// Per TanStack's recommendation:
+// https://tanstack.com/table/v8/docs/guide/row-models#customizefork-row-models
+
+function groupBy<TData extends RowData>(rows: Row<TData>[], columnId: string) {
+    const groupMap = new Map<any, Row<TData>[]>();
+
+    return rows.reduce((map, row) => {
+        // This line changed from original since getGroupingValue does not exist here as it should.
+        // const resKey = `${row.getGroupingValue(columnId)}`
+        const resKey = (row.original as any)[columnId].value.raw;
```

Comment:
can these `any` be `unknown` (without a lot of changes)?

Reply:
Yes, looks OK.

Reply:
perfect, thanks!